src/FileProviders/Physical/src/PhysicalFileProvider.cs

```diff
@@ -84,21 +84,19 @@ public PhysicalFileProvider(string root, ExclusionFilters filters)
         /// </value>
         public bool UsePollingFileWatcher
         {
-            get
+            get => _usePollingFileWatcher ?? false;
+            set
             {
                 if (_fileWatcher != null)
                 {
                     throw new InvalidOperationException($"Cannot modify {nameof(UsePollingFileWatcher)} once file watcher has been initialized.");
                 }
-
                 if (_usePollingFileWatcher == null)
                 {
                     ReadPollingEnvironmentVariables();
```

Comment:
I think we need to keep this `if` clause in the `get` accessor. It's just the exception that needs to move.